jerry-core/ecma/builtin-objects/ecma-builtin-helpers-sort.c

```diff
@@ -136,11 +146,15 @@ ecma_builtin_helper_array_heap_sort_helper (ecma_value_t *array_p, /**< array to
     array_p[i] = swap;
 
     /* Rebuild binary tree from the remaining elements. */
-    ECMA_TRY_CATCH (value,
-                    ecma_builtin_helper_array_to_heap (array_p, 0, i - 1, compare_func, sort_cb),
-                    ret_value);
-    ECMA_FINALIZE (value);
+    ecma_value_t value = ecma_builtin_helper_array_to_heap (array_p, 0, i - 1, compare_func, sort_cb);
+
+    if (ECMA_IS_VALUE_ERROR (value))
+    {
+      return value;
+    }
+
+    ecma_free_value (value);
```

Comment:
`ecma_builtin_helper_array_to_heap` should return with empty or error value, so freeing it is unnecessary.